addons/misra.py

```diff
@@ -256,33 +256,46 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand2]
 
 def getWhileLoopExpressions(whileToken):
-    if not whileToken or whileToken.str != 'while':
+    counter = []
+    if not simpleMatch(whileToken, 'while ('):
         return None
     lpar = whileToken.next
-    if not lpar or lpar.str != '(':
-        return None
     if not lpar.astOperand2.astOperand1:
         return None
-    counter = lpar.astOperand2.astOperand1.str
-    token = lpar
-    while (token and token.str != "{"):
-        token = token.next
-    token_link = token
-    if token_link:
-        while (token and token != token_link.link):
+    rpar = lpar.link
+    counter_token = lpar.next
+    while (counter_token != rpar):
```

Comment:
I would not loop like this. Now if the condition is:

    while (abc.a < 10)

... then as I read it, the counter will be "abc"

How do you think this should be handled:

    while (x < y)

... Do you think that `x` is a counter? Do you think that `y` is a counter? I think that either both should be counters or none should be counters.

I would recursively look at the condition from the parent..

    def findCounterTokens(cond):
        if cond.str in ['&&' , '||']:
            c = findCounterTokens(cond.astOperand1)
            c.extend(findCounterTokens(cond.astOperand2))
            return c
        ret = []
        if cond.isComparisonOp and cond.astOperand1.isName and cond.astOperand2.isNumber:
             ret.append(cond.astOperand1)
        elif cond.isComparisonOp and cond.astOperand2.isName and cond.astOperand1.isNumber:
             ret.append(cond.astOperand2)
        return ret

and it will be called like this:

    counter_tokens = findCounterTokens(lpar.astOperand2)